src/braket/circuits/gates.py

```diff
@@ -39,16 +40,31 @@
     3. Register the class with the `Gate` class via `Gate.register_gate()`.
 """
 
+
+class _FiniteOrderGate(Gate, ABC):
+    def __init__(self, order: int, qubit_count: Optional[int], ascii_symbols: Sequence[str]):
+        super().__init__(qubit_count, ascii_symbols)
+        self._order = order
+
+    def adjoint_list(self) -> List[Gate]:
+        return [self for _ in range(self._order - 1)]
+
+
+class _HermitianGate(_FiniteOrderGate, ABC):
+    def __init__(self, qubit_count: Optional[int], ascii_symbols: Sequence[str]):
+        super().__init__(2, qubit_count, ascii_symbols)
+
+
 # Single qubit gates #
 
 
-class H(Gate):
+class H(_HermitianGate):
     """Hadamard gate."""
 
     def __init__(self):
         super().__init__(qubit_count=None, ascii_symbols=["H"])
 
-    def to_ir(self, target: QubitSet):
+    def _to_ir(self, target: QubitSet):
```

Comment:
Can we just change all these to private? Is that a breaking change?

Reply:
Won't be breaking because they all inherit from `Gate`, where the public `to_ir` is defined.